ledger/src/blockstore.rs

```diff
@@ -1263,6 +1284,10 @@ impl Blockstore {
         if result {
             index_meta_working_set_entry.did_insert_occur = true;
             metrics.num_inserted += 1;
+
+            merkle_root_metas
+                .entry(erasure_set)
+                .or_insert(MerkleRootMeta::from_shred(&shred));
```

Comment:
Why do we need the code in L1204 then?

I guess follow up commits will compare if the merkle-roots are the same.
but then I guess we could read the blockstore-db here instead of pre-emptively fetching it but then not using it if the shred is not inserted.

Reply:
so the idea is:

L1204 - load merkle root meta from blockstore -> in memory map
(future commit) L1196 - compare with the current merkle shred meta for conflict
(here) L1287 - if the in memory map is empty and the insert has succeeded, add the current merkle shred meta to the in memory map
L1024 - commit in memory map to blockstore

Reply:
> but then I guess we could read the blockstore-db here instead of pre-emptively fetching it but then not using it if the shred is not inserted.

this is possible, but it would result in 2 blockstore reads, one for the future comparison, and then one to decide whether to insert the newly received merkle root meta.